content/en/contribute/code/core/dev-environment.md

```diff
@@ -18,27 +20,46 @@ These steps apply to both 3.x and 4.x CHT core development, unless stated otherw
 
 ## The Happy Path Installation
 
-This CHT Core developer guide will have you install NodeJS, npm, Grunt and CouchDB (via Docker) on your local workstation. These instructions should work verbatim on Ubuntu 18-22 (see [Ubuntu 18 note](#ubuntu-1804)), but will need tweaks for MacOS (via `brew`, see [MacOS > 12.3 note](#macos--123)) or Windows (via WSL2).
+CHT Core development can be done on Linux, macOS, or Windows (using the [Windows Subsystem for Linux (WSL2)](https://learn.microsoft.com/en-us/windows/wsl/install)). This CHT Core developer guide will have you install NodeJS, npm, Grunt and CouchDB (via Docker) on your local workstation.
 
 ### Install NodeJS, npm, grunt and Docker
 
-First, update your current Ubuntu packages and install some supporting tools via `apt`:
+First, update your current packages and install some supporting tools:
 
-```shell
+_(Node {{< param nodeVersion >}} is the environment used to run the CHT server in production, so this is the recommended version of Node to use for development.)_
+
+{{< tabpane persistLang=false lang=shell >}}
+{{< tab header="Linux (Ubuntu)" >}}
 sudo apt update && sudo apt -y dist-upgrade
 sudo apt -y install xsltproc curl uidmap jq python2 git make g++
-```
-
-Then install `nvm`, add it to your path and install NodeJS 16:
-
-```shell
+# Use NVM to install NodeJS:
 export nvm_version=`curl -s https://api.github.com/repos/nvm-sh/nvm/releases/latest | jq -r .name`
-curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/$nvm_version/install.sh | $0
-. ~/.$0rc
-nvm install 16
-```
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/$nvm_version/install.sh | $SHELL
+. ~/.$(basename $SHELL)rc
+nvm install {{< param nodeVersion >}}
+{{< /tab >}}
+{{< tab header="macOS" >}}
+# Uses Homebrew: https://brew.sh/
+brew update
+brew install curl jq pyenv git make node@{{< param nodeVersion >}} gcc
+# Python no longer included by default in macOS >12.3 
+pyenv install 2.7.18
+pyenv global 2.7.18
+echo "eval \"\$(pyenv init --path)\"" >> ~/.$0rc
+. $0
```

Comment:
I received the following failure when I ran the `. $0`
``` bash
➜  ~ echo "eval \"\$(pyenv init --path)\"" >> ~/.$0rc
➜  ~ . $0
.: no such file or directory: -zsh
```

I was reading [the comment](https://github.com/medic/cht-docs/pull/789/files#r1132726714) that @mrjones-plip added, and I am seeing differences between `$(basename $SHELL)`  and `$0`

``` bash
➜  ~ echo $0
-zsh
➜  ~ echo $(basename $SHELL)
zsh
```

I already discussed this with @jkuester, I am just leaving the comment here for visibility.

Reply:
Many thanks for double-checking this behavior!  I have switched all of this to use `$SHELL` for now.